test/testdata/lsp/requires_ancestor.rb

```diff
@@ -0,0 +1,18 @@
+# typed: true
+# enable-experimental-requires-ancestor: true
```

Comment:
Is this used by the test suite?

Reply:
Yes, for LSP test runner: https://github.com/sorbet/sorbet/pull/3409/files#diff-9a281b58f3ec5ea0244c2ac093f3298a22f1fe00828e2ff2e4cefb468ac94ebeR323

I also added them to the other tests so the `pipeline_test_runner` enables the option only when needed :+1